postpic/datareader/epochsdf.py

```diff
@@ -84,7 +84,7 @@ def data(self, key):
     def gridoffset(self, key, axis):
         axid = helper.axesidentify[axis]
         dx = self.gridspacing(key, axis)
-        if self[key].stagger & (1 << axid):
+        if hasattr(self[key], 'stagger') and (self[key].stagger & (1 << axid)):
```

Comment:
![screenshot_20171107_100711](https://user-images.githubusercontent.com/7170853/32485345-6cdf0978-c3a3-11e7-8bf5-677e971f4d8e.png)
In the screenshot `d` refers to the dimension of the code and `sims[d]` is a `simreader` for a d-dimensional simulation. Looking at the 1th dump of each simulation and each of the 6 fields, the `stagger` attribute is fixed. The value of stagger is fixed for the different fields. For older dumps without stagger attribute we can just pick the values from this list.

Reply:
From the `SDF` code (`sdf_common.f90`):
```
  INTEGER(i4), PARAMETER :: c_stagger_cell_centre = 0
  INTEGER(i4), PARAMETER :: c_stagger_face_x = 1
  INTEGER(i4), PARAMETER :: c_stagger_face_y = 2
  INTEGER(i4), PARAMETER :: c_stagger_face_z = 4
  INTEGER(i4), PARAMETER :: c_stagger_edge_x = &
      c_stagger_face_y + c_stagger_face_z
  INTEGER(i4), PARAMETER :: c_stagger_edge_y = &
      c_stagger_face_x + c_stagger_face_z
  INTEGER(i4), PARAMETER :: c_stagger_edge_z = &
      c_stagger_face_x + c_stagger_face_y
  INTEGER(i4), PARAMETER :: c_stagger_vertex = &
      c_stagger_face_x + c_stagger_face_y + c_stagger_face_z
```

From the epoch code (`shared_data.F90`):
```
  USE sdf, ONLY : c_stagger_edge_x, c_stagger_edge_y, c_stagger_edge_z, &
      c_stagger_face_x, c_stagger_face_y, c_stagger_face_z, &
      c_stagger_cell_centre, c_stagger_vertex, c_id_length
[...]
  INTEGER, PARAMETER :: c_stagger_ex = c_stagger_face_x
  INTEGER, PARAMETER :: c_stagger_ey = c_stagger_face_y
  INTEGER, PARAMETER :: c_stagger_ez = c_stagger_face_z
  INTEGER, PARAMETER :: c_stagger_bx = c_stagger_edge_x
  INTEGER, PARAMETER :: c_stagger_by = c_stagger_edge_y
  INTEGER, PARAMETER :: c_stagger_bz = c_stagger_edge_z
```
`diagnostics.F90`:
```
      CALL write_field(c_dump_ex, code, 'ex', 'Electric Field/Ex', 'V/m', &
          c_stagger_ex, ex)
      CALL write_field(c_dump_ey, code, 'ey', 'Electric Field/Ey', 'V/m', &
          c_stagger_ey, ey)
      CALL write_field(c_dump_ez, code, 'ez', 'Electric Field/Ez', 'V/m', &
          c_stagger_ez, ez)

      CALL write_field(c_dump_bx, code, 'bx', 'Magnetic Field/Bx', 'T', &
          c_stagger_bx, bx)
      CALL write_field(c_dump_by, code, 'by', 'Magnetic Field/By', 'T', &
          c_stagger_by, by)
      CALL write_field(c_dump_bz, code, 'bz', 'Magnetic Field/Bz', 'T', &
          c_stagger_bz, bz)
```
And `write_field` just passes the stagger variable on to `sdf_write_plain_variable`. Nothing here is dynamic, the values are always the same.

Reply:
Also some more default stagger values are available for e.g. the currents, just look into `diagnostics.F90` and `shared_data.F90`.